bitcoin/src/Bitcoin/Transaction/Taproot.hs

```diff
@@ -249,25 +228,27 @@ encodeTaprootWitness = \case
 -- | Verify that the script path spend is valid, except for script execution.
 verifyScriptPathData ::
     -- | Output key
-    PubKey ->
+    PubKeyXY ->
     ScriptPathData ->
     Bool
 verifyScriptPathData outputKey scriptPathData = fromMaybe False $ do
-    tweak commitment >>= fmap onComputedKey . tweakAddPubKey (scriptPathInternalKey scriptPathData)
+    tweak <- importTweak commitment
+    tweaked <- pubKeyXOTweakAdd (scriptPathInternalKey scriptPathData) tweak
+    pure $ uncurry onComputedKey . xyToXO $ tweaked
   where
-    onComputedKey computedKey =
-        XOnlyPubKey outputKey == XOnlyPubKey computedKey
-            && expectedParity == keyParity computedKey
+    onComputedKey computedKey computedParity =
+        fst (xyToXO outputKey) == computedKey
+            && expectedParity == computedParity
```

Comment:
This is probably the trickiest part and could make the difference between neutering the test and updating it correctly.